File: projectEuler/30/30.cpp
//Digit Fifth Powers
//Find the sum of all the numbers that can be written as the sum of fifth powers of their digits.

#include <iostream>
#include <cmath>
#include <vector>

using namespace std;

int main(){
    vector<int> v;
    for (int i = 2; i < 1000000; i++) {
        int sum = 0;
        int temp = i;
        while (temp > 0) {
            sum += pow(temp % 10, 5);
            temp /= 10;
        }
        if (sum == i) {
            v.push_back(i);
        }
    }
}

File: projectEuler/30/30.py
#Digit Fifth Powers
#Find the sum of all the numbers that can be written as the sum of fifth powers of their digits.

def digitFifthPowers():
    fifthPowers = []
    for i in range(2,1000000):
        if i == sum([int(x)**5 for x in str(i)]):
            fifthPowers.append(i)
    return sum(fifthPowers)

print(digitFifthPowers())